Documentation/ipvlan.md

```diff
@@ -38,3 +38,40 @@ Because all ipvlan interfaces share the MAC address with the host interface, DHC
 Therefore the container will not be able to reach the host via `ipvlan` interface.
 Be sure to also have container join a network that provides connectivity to the host (e.g. `ptp`).
 * A single master interface can not be enslaved by both `macvlan` and `ipvlan`.
+
+### Using ipvlan and dhcp
+There are two considerations when using "ipvlan" plugin together with "dhcp" IPAM plugin.
+
+First, since ipvlan driver uses the destination address in the IP packet when deciding to which slave the packet should be forwarded to, it will not work if the container interface is not yet assigned an IP.
+Since the DHCP packets are exchanged to determine which IP should be used, the interface will not yet have an IP address.
+A DHCP server typically unicasts the replies back to the client which will cause the ipvlan driver to drop the replies.
+To work around this, the DHCP client needs to instruct the DHCP server to use broadcasts instead.
+This can be accomplished by setting the "broadcast" flag in the "ipam" section of the network configuration:
+
+```
+{
+	"name": "mynet",
+	"type": "ipvlan",
+	"master": "eth0",
+	"ipam": {
+		"type": "dhcp",
+		"broadcast": true
+	}
+}
+```
+
+Second, because all ipvlan interfaces share the MAC address of the master interface, DHCP servers will have trouble identifying and tracking individual container interfaces.
+To combat this, it is necessary to include DHCP option 61 (Client Identifier) in DHCP messages.
+DHCP plugin will do so if "sendClientID" flag is enabled (default).
+The value of Client Identifier option will be ContainerID plus the network name.
+In addition, the DHCP server needs to echo option 61 (Client Identifier) in its replies.
+Not all DHCP servers are capable of doing so while others need to be explicitly configured to enable this behavior.
```

Comment:
Not all DHCP servers are capable of properly echoing the Client Identifier, while others may need to be explicitly configured to enable this behavior.